### PR TITLE
Home Assistant - Climate - Expose local temperature as a sensor

### PR DESCRIPTION
**Pull request by frenetic00**
Exposing the local temperature as a sensor allows temperature statistics to be tracked in home assistant

### CODE REVIEW

**Comment by Koenkk** [comment]
This creates duplicates data in HA and I'm not sure if others are also interested. Therefore holding this back until more people are interested in this.

**Comment by stefanuytterhoeven** [comment]
> This creates duplicates data in HA and I'm not sure if others are also interested. Therefore holding this back until more people are interested in this.

Interested! Now, as far as I know, we need to create templates in ha to accomplish this.
I have several 'multi sensors ' with temperature, humidity and pressure in one device. For me, it would be nice if they all would appear as separate sensors in ha. But, yeah, don't know if everyone feels the same

**Comment by ArtCore7** [comment]
Would Love to see this integrated soon...

**Comment by novisys** [comment]
Interested to 💯

**Comment by tijmenvdk** [comment]
Yes, this would be helpful, currently the datapoints can be used in HA as attribute values, but the option to have them as separate sensors would be great.

File: lib/extension/homeassistant.ts [hunk 1]
```diff
@@ -234,6 +234,19 @@ export default class HomeAssistant extends Extension {
                 },
             };
 
+            const localTempEntry: DiscoveryEntry = {
+                type: 'sensor',
+                object_id: endpoint ? `${temperature.name}_${endpoint}` : `climate_${temperature.name}`,
```
Comment (Koenkk) [inline lib/extension/homeassistant.ts:239]: Can you change `climate_${temperature.name}` -> `${temperature.name}` (I'm not sure if this was done on purpose)

Comment (frenetic00) [reply lib/extension/homeassistant.ts:239]: When I change `climate_${temperature.name}` -> `${temperature.name}` the code fails a test when I run
`npm run test-with-coverage`

I get the error:
`HomeAssistant extension > Should not have duplicate type/object_ids in mapping`

Comment (Koenkk) [reply lib/extension/homeassistant.ts:239]: I think that is due to some devices having `e.local_temperature()` as a separate sensor. Can you try removing this and see if the issue is gone? 

e.g. 

- https://github.com/Koenkk/zigbee-herdsman-converters/blob/358793bc1e9d9af06ae508c1b496532d644de014/devices/stelpro.js#L39 (more occurrences in this file)
- https://github.com/Koenkk/zigbee-herdsman-converters/blob/358793bc1e9d9af06ae508c1b496532d644de014/devices/namron.js#L382

Comment (frenetic00) [reply lib/extension/homeassistant.ts:239]: Thanks for pointing in me in the right direction.
I'll take a look

**Comment by hugobloem** [comment]
Although this would be an interesting thing to have. I do think this is a problem with the Home Assistant climate implementation and not necessarily with Z2M. If you do want the statistics you could make a template sensor from the climate attribute.

**Comment by rsporsche** [comment]
I'm interested in this because I overrode the HA climate 'current_temperature_template' to use the externally measured value and now I don't get the local temperature in HA.

**Pull request closed without merging**